tools/test-npm.sh

```diff
@@ -5,13 +5,26 @@ set -e
 # always change the working directory to the project's root directory
 cd $(dirname $0)/..
 
-# pass NODE_EXE from something like Makefile
-# it should point to either {./}node or {./}node.exe, depending on the platform
-if [ -z $NODE_EXE ]; then
+# pass a $NODE environment variable from something like Makefile
+# it should point to either ./{node} or ./{node}.exe, depending on the platform
+if [ -z $NODE ]; then
   echo "No node executable provided. Bailing." >&2
   exit 0
 fi
 
+
+## a hopefully temporary hack to use the local $NODE binary when calling node(1)
+# make a link to $NODE as ./node
+ln -fs $NODE node
+# get the absolute path we are in -- see: http://stackoverflow.com/a/4774063/1279026
+pushd `dirname $0` > /dev/null
+PWD=`pwd -P`
+popd > /dev/null
+# export that link as preferred in the $PATH
+export PATH=$PWD:$PATH
```

Comment:
Hmmm, nope. It appears the tests themselves depend on a global `node` (from the tap module, I think?)

Since running the test requires it, and causes it to fail, that's not a possibility.